### DISCUSSION TITLE
How to properly implement search feature

### DISCUSSION THREAD

**Discussion by 1alexvash** [vercel/next.js#40631] [Help]
I got a website:
https://alex-code.com/
https://github.com/1alexvash/AlexCodeBlog

The feature works on localhost, but it does not work on staging / deployment
![animation](https://user-images.githubusercontent.com/46556479/190841122-58f9b96a-037f-49ef-b2de-b2ccde3c31fc.gif)

The way it works locally: I created this beautiful NextJS function which goes and reads the posts from the _/projectName/content/posts folder_
![image](https://user-images.githubusercontent.com/46556479/190841338-b4573c46-67ee-469f-8379-5b02ae820fea.png)

And it works like the charm, but the thing is, this folder does not make it to the Next deploy bundle, so after deploy I don't really have access to it

You can verify by going to https://alex-code.com/api/getAllPosts , it returns empty array only without posts data

--------

So my question is how to implement this feature properly?

It looks like I went with Client Side Fetching pattern, but it didn't work for me 😑

**Comment by destructo570** [comment]
Did you try to serve that content folder statically from a public folder ?

Next.js can serve static files, like images, under a folder called public in the root directory. Files inside public can then be referenced by your code starting from the base URL (/).

https://nextjs.org/docs/basic-features/static-file-serving

May be this will help.

**Comment by 1alexvash** [comment]
@destructo570 That's one sure way to got about it

But what if I don't want to put inside the **/public** folder?

I was just thinking maybe I'm not using getServerSideProps or getStatisProps feature properly

---

Because when I'm visiting any blog page like:
https://alex-code.com/post/2022-08-05-book-review-so-good-they-cant-ignore-you
And when I go to the Network tab, I'm seeing the full list of my posts:
![Untitled](https://user-images.githubusercontent.com/46556479/190886930-c2d2747d-6844-4b30-ab14-a4be2294934e.png)

Though here comes another peculiarity, this doesn't happen locally 🤯

**Comment by 1alexvash** [comment]
Um, apparently those props above were coming from NextJS getStaticProps, so that's why I got them on the picture above

**Comment by 1alexvash** [comment]
I fixed my feature by taking code from here:
https://github.com/leerob/nextjs-prism-markdown/blob/main/lib/docs.js
For some reason this markdown reader, has the access to the markdown files outside public folder on staging
🤯
